### PR TITLE
Remove handling account ID in app, prevent changing settings when signed in

### LINKED ISSUE #2668: Disable changing settings when signed in
Changing the settings while signed in causes the tunnel to go down (crash?).

Then, if I sign out with the Settings window still open, the Auth Base URL resets to the staging URL, and the API URL resets to the production URL.

It would probably make sense to prevent users from changing these while signed in. Or, we could allow changing them, and then warn the user upon hitting Apply that it will sign them out.

### PR DESCRIPTION
**Pull request by roop**
This PR

  - Addresses the Apple clients part of #2791 
  - Fixes #2668

The Settings UI change of separating Logs and Advanced settings is not part of this PR.

### CODE REVIEW

File: swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift [hunk 1]
```diff
@@ -176,95 +164,50 @@ public struct SettingsView: View {
...
+        "Saving settings will sign you out",
+        isPresented: $isShowingConfirmationAlert,
+        presenting: confirmationAlertContinueAction,
+        actions: { confirmationAlertContinueAction in
+          Button("Continue") {
+            confirmationAlertContinueAction.performAction(on: self)
+          }
+          Button("Cancel", role: .cancel) {
+            // Nothing to do
+          }
```
Comment (jamilbk) [inline swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift]: Does order matter here?

```suggestion
          Button("Cancel", role: .cancel) {
            // Nothing to do
          }
          Button("Continue") {
            confirmationAlertContinueAction.performAction(on: self)
          }
```

File: swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift [hunk 2]
```diff
@@ -176,95 +164,50 @@ public struct SettingsView: View {
...
```
Comment (jamilbk) [inline swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift]:
```suggestion
          Button("Cancel", role: .cancel) {
            // Nothing to do
          }
          Button("Continue") {
            confirmationAlertContinueAction.performAction(on: self)
          }
```

File: swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/TunnelStore.swift [hunk 3]
```diff
@@ -12,11 +12,14 @@ import OSLog
...
   case tunnelCouldNotBeStopped
+  case cannotSaveToTunnelWhenConnected
+  case cannotSignOutWhenConnected
+  case stopAlreadyBeingAttempted
 }
 
 public struct TunnelProviderKeys {
   static let keyAuthBaseURLString = "authBaseURLString"
-  static let keyAccountId = "accountId"
+  static let keyAccountId = "accountId"  // Unused
```
Comment (jamilbk) [inline swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/TunnelStore.swift]: Can this just be removed if we're not using it?
```suggestion
```

**Review by jamilbk — CHANGES_REQUESTED** [review]
Just a question on the button order and minor cleanup otherwise LGTM

File: swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift (hunk 1 above)
Comment (roop) [reply swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift]: It doesn't. SwiftUI will take care of placing the cancel-role-button to the left. Nevertheless, I agree it's a good idea to change the order in the code anyway.

File: swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/TunnelStore.swift (hunk 3 above)
Comment (roop) [reply swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/TunnelStore.swift]: We use it to remove the data from the tunnel configuration for tunnel configurations created by older versions of the app. I will clarify the comment.

File: swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift (hunk 1 above)
Comment (roop) [reply swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift]: Fixed in commit c9d6145.

File: swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift (hunk 2 above)
Comment (roop) [reply swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift]: Fixed in commit c9d6145.

File: swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/TunnelStore.swift (hunk 3 above)
Comment (roop) [reply swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/TunnelStore.swift]: Fixed in commit e2e79d7.

**Review by jamilbk — APPROVED** [review]

**Pull request merged**